src/base-widget.ts

```diff
@@ -3,8 +3,16 @@ import { isPlatformBrowser } from '@angular/common';
 import { bem, noop } from './utils';
 
 export class Widget {
-  public init: () => void;
-  public getConfiguration: () => object;
+  public init: (
+    params: {
+      templatesConfig: object;
+      state: object;
+      results: {}[];
```

Comment:
If we want to be consistent:

```suggestion
      results: object[];
```